helpers/browser/driver.js

```diff
@@ -232,6 +238,59 @@ class ChromeProtocol {
       });
     });
   }
+
+  beginNetworkCollect() {
+    return this.getInstance().then(chrome =>
+      new Promise((resolve, reject) => {
+        this._networkRecords = [];
+        this._networkRecorder = new NetworkRecorder(this._networkRecords);
+
+        chrome.on('Network.requestWillBeSent',
```

Comment:
per discussion, punting on this as driver.js evolves and we figure out its life cycle better. Just kind of ugly for now, but the event registration/deregistration are still contained entirely within these two methods